packages/aws-cdk-lib/aws-route53/lib/health-check.ts

```diff
@@ -0,0 +1,658 @@
+    ruleHealthThresholdIsNotAllowed,
+    validateFqdn,
+    validateIpAddress,
+    validateEitherFqdnOrIpAddressRequired,
+    validateRequestInterval,
+    validateFailureThreshold,
+    validateRegions,
+  ],
+  [HealthCheckType.CLOUDWATCH_METRIC]: [
+    ruleAlarmIdentifierIsRequired,
+    rulePortIsNotAllowed,
+    ruleEnableSNIIsNotAllowed,
+    ruleSearchStringIsNotAllowed,
+    ruleChildHealthChecksIsNotAllowed,
+    ruleRoutingControlIsNotAllowed,
+    ruleHealthThresholdIsNotAllowed,
+    ruleRequestIntervalIsNotAllowed,
+    ruleFailureThresholdIsNotAllowed,
+    ruleRegionsIsNotAllowed,
+  ],
+  [HealthCheckType.CALCULATED]: [
+    validateChildHealthChecks,
+    ruleAlarmIdentifierIsNotAllowed,
+    rulePortIsNotAllowed,
+    ruleEnableSNIIsNotAllowed,
+    ruleSearchStringIsNotAllowed,
+    ruleRoutingControlIsNotAllowed,
+    ruleRequestIntervalIsNotAllowed,
+    ruleFailureThresholdIsNotAllowed,
+    ruleRegionsIsNotAllowed,
+  ],
+  [HealthCheckType.RECOVERY_CONTROL]: [
+    ruleAlarmIdentifierIsNotAllowed,
+    ruleRoutingControlIsRequired,
+    ruleEnableSNIIsNotAllowed,
+    ruleSearchStringIsNotAllowed,
+    ruleChildHealthChecksIsNotAllowed,
+    rulePortIsNotAllowed,
+    ruleHealthThresholdIsNotAllowed,
+    ruleRequestIntervalIsNotAllowed,
+    ruleFailureThresholdIsNotAllowed,
+    ruleRegionsIsNotAllowed,
+  ],
+};
+
+function getDefaultResourcePathForType(type: HealthCheckType): string | undefined {
+  switch (type) {
+    case HealthCheckType.HTTP:
+    case HealthCheckType.HTTP_STR_MATCH:
+    case HealthCheckType.HTTPS:
+    case HealthCheckType.HTTPS_STR_MATCH:
+      return '';
+    default:
+      return undefined;
+  }
+}
+
+function getDefaultInsufficientDataHealthStatusForType(type: HealthCheckType): InsufficientDataHealthStatusEnum | undefined {
+  if (type === HealthCheckType.CLOUDWATCH_METRIC) {
+    return InsufficientDataHealthStatusEnum.LAST_KNOWN_STATUS;
+  }
+
+  return undefined;
+}
+
+function getDefaultMeasureLatencyForType(type: HealthCheckType): boolean | undefined {
+  switch (type) {
+    case HealthCheckType.CALCULATED:
+    case HealthCheckType.CLOUDWATCH_METRIC:
+      return undefined;
+    default:
+      return false;
+  }
+}
+
+function getDefaultHealthThresholdForType(props: HealthCheckProps): number | undefined {
+  if (props.type === HealthCheckType.CALCULATED) {
+    return props.childHealthChecks!.length;
+  }
+
+  return undefined;
+}
+
+function getDefaultFailureThresholdForType(type: HealthCheckType): number | undefined {
+  switch (type) {
+    case HealthCheckType.CALCULATED:
+    case HealthCheckType.CLOUDWATCH_METRIC:
+      return undefined;
+    default:
+      return 3;
+  }
+}
+
+function getDefaultRequestIntervalForType(type: HealthCheckType): Duration | undefined {
+  switch (type) {
+    case HealthCheckType.CALCULATED:
+    case HealthCheckType.CLOUDWATCH_METRIC:
+      return undefined;
+    default:
+      return Duration.seconds(30);
```

Comment:
same as above.

Reply:
hi @GavinZZ , please take a look on my another response to similar comment